controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -145,6 +147,67 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
 
+		consumerConfigMap := &corev1.ConfigMap{}
+		consumerConfigMap.Name = r.storageConsumer.Name
+		if r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name != "" {
+			consumerConfigMap.Name = r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name
+		}
+		consumerConfigMap.Namespace = r.namespace
+
+		if err := r.get(consumerConfigMap); client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, err
+		} else if consumerConfigMap.UID == "" {
+			cephCluster, err := util.GetCephClusterInNamespace(r.ctx, r.Client, r.namespace)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if cephCluster.Status.CephStatus == nil || cephCluster.Status.CephStatus.FSID == "" {
+				return reconcile.Result{}, fmt.Errorf("waiting for Ceph FSID")
+			}
+			clientProfileName := util.CalculateMD5Hash([2]string{cephCluster.Status.CephStatus.FSID, r.storageConsumer.Name})
+			if len(clientProfileName) > 36 {
+				// truncating at 36 as that is the maximum limit that csi can use in it's handle
+				clientProfileName = clientProfileName[:36]
+			}
+			//Create the configMap
+			consumerConfigMap.Data = map[string]string{}
+			consumerResources := util.WrapStorageConsumerResourceConfigMap(consumerConfigMap)
+			consumerResources.SetRbdRadosNamespaceName(r.storageConsumer.Name)
+			consumerResources.SetSubVolumeGroupName(r.storageConsumer.Name)
+			consumerResources.SetSubVolumeGroupRadosNamespaceName(r.storageConsumer.Name)
+			consumerResources.SetRbdClientProfileName(clientProfileName)
+			consumerResources.SetCephFsClientProfileName(clientProfileName)
+			consumerResources.SetNfsClientProfileName(clientProfileName)
+			consumerResources.SetCsiRbdNodeSecretName(fmt.Sprintf("rbd-node-%s", clientProfileName))
+			consumerResources.SetCsiRbdProvisionerSecretName(fmt.Sprintf("rbd-provisioner-%s", clientProfileName))
+			consumerResources.SetCsiCephFsNodeSecretName(fmt.Sprintf("cephfs-node-%s", clientProfileName))
+			consumerResources.SetCsiCephFsProvisionerSecretName(fmt.Sprintf("cephfs-provisioner-%s", clientProfileName))
+			consumerResources.SetCsiNfsNodeSecretName(fmt.Sprintf("nfs-node-%s", clientProfileName))
+			consumerResources.SetCsiNfsProvisionerSecretName(fmt.Sprintf("nfs-provisioner-%s", clientProfileName))
+			err = r.Client.Create(r.ctx, consumerConfigMap)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{Requeue: true}, err
+		} else if !controllerutil.HasControllerReference(consumerConfigMap) {
```

Comment:
How come this is an else? You need to add a controller ref even when creating a new config map.
Preferably you need to run this code and figure our if you are the new primary before reconciling the configmap content

Also, `HasControllerReference` is not enough, you also want to check if it is you or some other consumer,